packages/plugin-legacy/index.js

```diff
@@ -263,7 +263,7 @@ function viteLegacyPlugin(options = {}) {
 
       // transform the legacy chunk with @babel/preset-env
       const sourceMaps = !!config.build.sourcemap
-      const { code, map } = loadBabel().transform(raw, {
+      const { code } = loadBabel().transform(raw, {
```

Comment:
I didn't find a way to merge source maps in `MagicString` api, maybe making babel plugin would be a better solution?